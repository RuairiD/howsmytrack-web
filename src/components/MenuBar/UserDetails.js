import React from "react";

import { Menu, Icon, Typography } from "antd";
import { Div } from "lemon-reset";

const UserDetails = ({ username, rating }) => {
    if (rating) {
        return (
            <Menu.Item disabled className="ant-menu-item">
                <Div style={{ display: "flex" }}>
                    <Typography.Text
                        ellipsis
                        style={{
                            marginRight: "auto",
                            marginTop: "auto",
                            marginBottom: "auto",
                        }}
                    >
                        {username}
                    </Typography.Text>
                    <Typography.Text
                        strong
                        style={{
                            marginLeft: "auto",
                            marginTop: "auto",
                            marginBottom: "auto",
                        }}
                    >
                        {rating.toFixed(2)}<Icon type="star" />
                    </Typography.Text>
                </Div>
            </Menu.Item>
        );
    }
    return (
        <Menu.Item disabled className="ant-menu-item">
            <Div style={{ display: "flex" }}>
                <Typography.Text
                    ellipsis
                    style={{
                        marginRight: "auto",
                        marginTop: "auto",
                        marginBottom: "auto",
                    }}
                >
                    {username}
                </Typography.Text>
            </Div>
        </Menu.Item>
    );
};

export default UserDetails;
