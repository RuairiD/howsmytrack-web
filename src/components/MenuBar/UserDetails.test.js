import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import UserDetails from "./UserDetails";

describe("UserDetails", () => {
    it("renders the user's username and rating rounded to 2 decimal places", () => {
        const wrapper = shallow(
            <UserDetails
                username="username"
                rating={4.321}
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders the user's username only, if no rating is available", () => {
        const wrapper = shallow(
            <UserDetails
                username="username"
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
