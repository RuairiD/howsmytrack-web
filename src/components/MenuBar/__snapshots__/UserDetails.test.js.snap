// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetails renders the user's username and rating rounded to 2 decimal places 1`] = `
<MenuItem
  className="ant-menu-item"
  disabled={true}
>
  <Div
    className=""
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <Text
      ellipsis={true}
      style={
        Object {
          "marginBottom": "auto",
          "marginRight": "auto",
          "marginTop": "auto",
        }
      }
    >
      username
    </Text>
    <Text
      strong={true}
      style={
        Object {
          "marginBottom": "auto",
          "marginLeft": "auto",
          "marginTop": "auto",
        }
      }
    >
      4.32
      <Icon
        type="star"
      />
    </Text>
  </Div>
</MenuItem>
`;

exports[`UserDetails renders the user's username only, if no rating is available 1`] = `
<MenuItem
  className="ant-menu-item"
  disabled={true}
>
  <Div
    className=""
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <Text
      ellipsis={true}
      style={
        Object {
          "marginBottom": "auto",
          "marginRight": "auto",
          "marginTop": "auto",
        }
      }
    >
      username
    </Text>
  </Div>
</MenuItem>
`;
